src/System.Web.HttpUtility/src/System/Web/HttpUtility.cs

```diff
@@ -30,9 +30,8 @@ public override string ToString()
                 {
                     sb.AppendFormat("{0}={1}&", keys[i], UrlEncode(this[keys[i]]));
                 }
-                if (sb.Length > 0)
-                    sb.Length--;
-                return sb.ToString();
+
+                return sb.ToString(0, sb.Length - 1);
```

Comment:
Is `return sb.ToString(0, sb.Length - 1);` more efficient than `sb.Length--; return sb.ToString();`?

Reply:
A quick comparison suggests that it's very marginally so, though I've no real preference on it; the change to my mind was just "adjust length so we'll return the right string, if necessary" to realising it was always necessary and changing it to "return the right string".